src/Sarif/RuleUtilities.cs

```diff
@@ -105,6 +106,12 @@ public static string NormalizeTemplateId(string ruleId, string templateId)
             {
                 templateId = templateId.Substring(ruleId.Length + 1);
             }
+
+            if (!string.IsNullOrEmpty(prefix) && templateId.StartsWith(prefix + "_", StringComparison.Ordinal))
```

Comment:
>string [](start = 17, length = 6)

looking at your tests, i don't believe this conditional is ever hit. prove me wrong in code coverage? or delete this conditional if i am proven right.

Reply:
@michaelcfanning: The conditional is indeed hit, and both the true and false cases are exercised, as I just demonstrated to myself by

1. commenting it out and watching my unit test fail on this assertion with "The given key was not present in the dictionary"
    ```
     skimmer.RichMessageTemplates["Pass"]
        .Should().Be(SkimmerBaseTestResources.TST0001_Rich_Pass);
    ```
2. setting a breakpoint and watching it hit both the `true` and `false` branches when expected.